### DISCUSSION TITLE
Are there any length constrained on `prefix` of the identifier?

### DISCUSSION THREAD

**Discussion by therealharpaljadeja** [celo-org/social-connect#249] [Help (Q&A)] [answered]
Does the `prefix` need to comply with any length constraint or any other constraints?

**Comment by eelanagaraj** [comment, marked as answer]
When adding new prefixes to `IdentifierPrefix` (exported from `@celo/identity`, defined in `@celo/base` for use in other packages), there isn't a strict length constraint, but we do expect prefixes to follow the format as described in the docstring, namely that a new prefix should either be the same as the equivalent DID method or the full name of the platform.

When using a prefix generally, one should use a prefix defined in `IdentifierPrefix` (or submit a PR for a prefix not already included there) in order to create interoperable identifiers.

**Reply by therealharpaljadeja** [reply]
In a situation where one wants to have a closed ecosystem meaning interoperability among a few apps only, they can have an prefix of their own choice, right?
